### PR TITLE
kernel: Enable virtio-fs in kata kernel

### PR DESCRIPTION
**Pull request by ganeshmaharaj**
This patches adds virtio-fs capability to the kata kernel along with
config changes to enable the same on kata by default. The system will
only be exercised when `shared_fs` is set to `virtio-fs` in the kata
configuration file. the default still remains to be 9p

Fixes: #387

Signed-off-by: Ganesh Maharaj Mahalingam <ganesh.mahalingam@intel.com>

### CODE REVIEW

File: kernel/patches/4.19.x/0002-Enable-memory-hotplug-using-probe-for-arm64.patch [hunk 1]
```diff
@@ -1,7 +1,7 @@
-From 074a6a9d83a4e790f892ef0fc91cdabbfbf26202 Mon Sep 17 00:00:00 2001
```
Comment (devimc) [inline kernel/patches/4.19.x/0002-Enable-memory-hotplug-using-probe-for-arm64.patch]: with this patch, now we have 3 patches that start with `0001`, would you mind changing their name?

Comment (ganeshmaharaj) [reply kernel/patches/4.19.x/0002-Enable-memory-hotplug-using-probe-for-arm64.patch]: I believe this change actually renames all the patches to have a proper order. Do you think i have missed it?

Comment (devimc) [reply kernel/patches/4.19.x/0002-Enable-memory-hotplug-using-probe-for-arm64.patch]: oops!, yes, you're right :smile:

File: kernel/configs/x86_64_kata_kvm_4.19.x [hunk 2]
```diff
@@ -1,13 +1,13 @@
 #
 # Automatically generated file; DO NOT EDIT.
-# Linux/x86 4.19.24 Kernel Configuration
+# Linux/x86 4.19.28 Kernel Configuration
 #
 
 #
-# Compiler: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
+# Compiler: gcc (Ubuntu 7.4.0-1ubuntu1~16.04~ppa1) 7.4.0
```
Comment (jodh-intel) [inline kernel/configs/x86_64_kata_kvm_4.19.x]: OOI, why are you using a ppa (non-standard) compiler version?

**Comment by jodh-intel** [comment]
We'd better get Mr Fragment^w^w @grahamwhaley to ack this one too... :smile:

File: kernel/configs/x86_64_kata_kvm_4.19.x (hunk 2 above)
Comment (grahamwhaley) [reply kernel/configs/x86_64_kata_kvm_4.19.x]: My guess - only way to get a recent gcc on bunty...

File: kernel/configs/x86_64_kata_kvm_4.19.x [hunk 3]
```diff
@@ -2597,7 +2597,9 @@ CONFIG_FANOTIFY=y
 # CONFIG_QUOTA is not set
 CONFIG_AUTOFS4_FS=y
 CONFIG_AUTOFS_FS=y
-# CONFIG_FUSE_FS is not set
+CONFIG_FUSE_FS=y
+# CONFIG_CUSE is not set
+CONFIG_VIRTIO_FS=y
```
Comment (grahamwhaley) [inline kernel/configs/x86_64_kata_kvm_4.19.x]: As these two `CONFIG` items are only present with the patches in this PR (that is, they are not upstream yet afaict), we can't apply them to the fragments PR until this PR has landed.
They look find in the context of this PR though.

**Comment by grahamwhaley** [comment]
This is ready to merge - we should probably bring this PR to the attention of @stefanha and @dagrh for a moment before we merge though :-)

**Comment by stefanha** [comment]
@grahamwhaley @ganeshmaharaj The config options look good.

**Comment by grahamwhaley** [comment]
Great, let's merge this puppy...

**Pull request merged**